src/middlewared/middlewared/common/attachment/__init__.py

```diff
@@ -138,10 +139,10 @@ async def restart_reload_services(self, attachments):
     async def remove_alert(self, attachment):
         await self.middleware.call(f'{self.namespace}.remove_locked_alert', attachment['id'])
 
-    async def is_child_of_path(self, resource, path, check_parent):
+    async def is_child_of_path(self, resource, path, check_parent, exact_match):
```

Comment:
Can you please add a comment that explains all the parameters of this method and how it should work? And a set of unit tests? It's getting complicated.